# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/AudioStream.h
# Commit: 30cd5bcfee92
# Full Hash: 30cd5bcfee929d4d1244c7135af0892614a553e4
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-05-24 21:44:48
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - Wait for the audio callbacks to start being called before using the audio clock. r=alwu,media-playback-reviewers
#   
#   Because we're generally using high latency on the cubeb stream used by
#   AudioStream instance, it can take some time for the callbacks to start being
#   called, and the for the audio clock (cubeb_stream_get_position(...)) to advance.
# ==============================================================================

diff -r acb4ae1c1e57 -r 30cd5bcfee92 dom/media/AudioStream.h
--- a/dom/media/AudioStream.h	Tue May 24 13:09:06 2022 +0000
+++ b/dom/media/AudioStream.h	Tue May 24 13:09:06 2022 +0000
@@ -297,6 +297,9 @@
 
   bool IsPlaybackCompleted() const;
 
+  // Returns true if at least one DataCallback has been called.
+  bool CallbackStarted() const { return mCallbacksStarted; }
+
  protected:
   friend class AudioClock;
 
@@ -386,6 +389,7 @@
   std::atomic<bool> mPreservesPitch;
   // Audio thread only
   bool mAudioThreadChanged = false;
+  Atomic<bool> mCallbacksStarted;
 };
 
 }  // namespace mozilla